### PR TITLE
Apply a few shell improvement in start scripts

### PR DESCRIPTION
**Pull request by apc999**
- Move `bootstrapConf` to `case` as other commands in `bin/alluxio`. The reason is we no more change `conf/alluxio-env.sh` inside `bootstrapConf`
- Add more Java version check in `libexec/alluxio-config.sh` --- we require Java 1.8 to run Alluxio at least for now
- Continue support of `ALLUXIO_UNDERFS_ADDRESS` and add warning about deprecation

### CODE REVIEW

File: bin/alluxio [hunk 1]
```diff
@@ -187,20 +186,16 @@ function main {
...
-  fi
-
   DEFAULT_LIBEXEC_DIR="${BIN}"/../libexec
   ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
   . ${ALLUXIO_LIBEXEC_DIR}/alluxio-config.sh
 
   PARAMETER=""
 
   case ${COMMAND} in
+  "bootstrapConf")
```
Comment (calvinjia) [inline bin/alluxio]: We are no longer supporting `bootstrap-conf` is that right?

Comment (apc999) [reply bin/alluxio]: right. I don't remember why we want to support `bootstrap-conf` in addition to `bootstrapConf`. do you recall?

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**